### PR TITLE
149 tabs should be routes

### PR DESCRIPTION
**Pull request by zpoken**

### CODE REVIEW

File: apps/webapp/app/ibc/constants.ts
Comment (grod220) [inline apps/webapp/app/ibc/constants.ts:1]: This should be in the constants package

**Review by grod220 — COMMENTED** [review]
- There appears to be a fade-in transition that is jarring. Please remove.
- The url does not match the nested nature of the design. Think we should have this:
  - `/dashboard` (goes to assets tab)
  - `/dashboard/transactions`
  - `/send` (goes to send tab)
  - `/send/receive` (kinda weird wording, but the best we have)
- Using nested routes will allow nextjs to not have to re-render the header

File: apps/webapp/hooks/typesafe-pathname.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { usePathname } from 'next/navigation';
+
+export const useTypesafePathname = <T>() => {
```
Comment (grod220) [inline apps/webapp/hooks/typesafe-pathname.ts:3]:
```diff
- useTypesafePathname
+ useTypedPathname
```

File: apps/webapp/app/header/navbar.tsx [hunk 2]
```diff
@@ -1,45 +1,51 @@
 'use client';
 
 import Link from 'next/link';
-import { usePathname } from 'next/navigation';
 import { DappPath } from './paths';
 import { cn } from '@penumbra-zone/ui/lib/utils';
+import { useTypesafePathname } from '../../hooks/typesafe-pathname';
 
 const links = [
```
Comment (grod220) [inline apps/webapp/app/header/navbar.tsx:8]: Please create an interface for this. Also, make `subLinks` an optional field so you don't need to do `subLinks: []` a lot.

File: apps/webapp/app/send/send-form.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ const InputToken = dynamic(() => import('../../shared/input-token'), {
   ssr: false,
 });
 
-export default function SendForm() {
+const Send = () => {
```
Comment (grod220) [inline apps/webapp/app/send/send-form.tsx:16]: Given this is dynamically imported, don't think you need to dynamically import `InputToken` above right?

**Review by grod220 — APPROVED** [review]
After those last changes, you can merge

File: apps/webapp/app/send/send-form.tsx (hunk 3 above)
Comment (Vadim121197) [reply apps/webapp/app/send/send-form.tsx:12]: Yes

**Pull request merged**